bentoml/yatai/repository/abs_repository.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from datetime import datetime, timedelta
+import os
+from pickle import NONE
+from urllib.parse import urlparse
+from dependency_injector.wiring import Provide, inject
+from bentoml.configuration import containers
+
+from bentoml.configuration.containers import BentoMLContainer
+from bentoml.exceptions import YataiRepositoryException
+from bentoml.yatai.proto.repository_pb2 import BentoUri
+from bentoml.yatai.repository.base_repository import BaseRepository
+
+logger = logging.getLogger(__name__)
+
+class ABSRepository(BaseRepository):
+    @inject
+    def __init__(
+        self,
+        base_url,
+        expiration: int = Provide[
+            BentoMLContainer.config.yatai.repository.abs.expiration
+        ],
+    ):
+        try:
+            from azure.storage.blob import (
+                BlobServiceClient,
+                generate_blob_sas,
+                AccountSasPermissions,
+            ) 
+        except ImportError:
+            raise YataiRepositoryException(
+                '"azure-storage-blob" package is required for Azure Blob Storage Repository.'
+                'You can install it with pip: '
+                '"pip install pip install azure-storage-blob"'
+                'Find out more at https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python'
+            )
+        self.generate_blob_sas = generate_blob_sas
+        self.sas_permission_add = AccountSasPermissions(
+            create=True,
+            write=True,
+        )
+        self.sas_permission_get = AccountSasPermissions(
+            read=True,
+        )
+
+        self.uri_type = BentoUri.ABS
```

Comment:
Can you make sure run `./proto/generate-docker.sh` to generate python and nodejs definitions? It doesn't work right now